GTG/core/dates.py

```diff
@@ -478,13 +477,17 @@ def _parse_only_month_day_for_recurrency(self, string, newtask=True):
         except ValueError:
             result = None
 
-        if result is None or result <= self_date:
+        if result is None or (not newtask and result <= self_date):
```

Comment:
I was going to but I felt that I would right a whole paragraph to explain it -_-.

So we need `not newtask` so that when it the first time the task is being enable it does not go to the next occurrence and when it's not the first, we make sure that the new date (result) is superior to `self_date` (to avoid an infinite loop).